go/embedded/online_features.go

```diff
@@ -25,26 +27,88 @@ type DataTable struct {
 	SchemaPtr uintptr
 }
 
-func NewOnlineFeatureService(conf *OnlineFeatureServiceConfig) *OnlineFeatureService {
+func NewOnlineFeatureService(conf *OnlineFeatureServiceConfig, transformationCallback feast.TransformationCallback) *OnlineFeatureService {
 	repoConfig, err := feast.NewRepoConfigFromJSON(conf.RepoPath, conf.RepoConfig)
 	if err != nil {
 		log.Fatalln(err)
 	}
 
-	fs, err := feast.NewFeatureStore(repoConfig)
+	fs, err := feast.NewFeatureStore(repoConfig, transformationCallback)
 	if err != nil {
 		log.Fatalln(err)
 	}
 
 	return &OnlineFeatureService{fs: fs}
 }
 
+func (s *OnlineFeatureService) GetEntityTypesMap(featureRefs []string) (map[string]int32, error) {
+	viewNames := make(map[string]interface{})
+	for _, featureRef := range featureRefs {
+		viewName, _, err := feast.ParseFeatureReference(featureRef)
+		if err != nil {
+			return nil, err
+		}
+		viewNames[viewName] = nil
+	}
+
+	entities, _ := s.fs.ListEntities(true)
+	entitiesByName := make(map[string]*feast.Entity)
+	for _, entity := range entities {
+		entitiesByName[entity.Name] = entity
+	}
+
+	joinKeyTypes := make(map[string]int32)
+
+	for viewName, _ := range viewNames {
+		view, err := s.fs.GetFeatureView(viewName, true)
+		if err != nil {
+			// skip on demand feature views
+			continue
+		}
+		for entityName, _ := range view.Entities {
+			entity := entitiesByName[entityName]
+			joinKeyTypes[entity.JoinKey] = int32(entity.ValueType.Number())
+		}
+	}
+
+	return joinKeyTypes, nil
+}
+
+func (s *OnlineFeatureService) GetEntityTypesMapByFeatureService(featureServiceName string) (map[string]int32, error) {
+	featureService, err := s.fs.GetFeatureService(featureServiceName)
+	if err != nil {
+		return nil, err
+	}
+
+	joinKeyTypes := make(map[string]int32)
+
+	entities, _ := s.fs.ListEntities(true)
+	entitiesByName := make(map[string]*feast.Entity)
+	for _, entity := range entities {
+		entitiesByName[entity.Name] = entity
+	}
+
+	for _, projection := range featureService.Projections {
+		view, err := s.fs.GetFeatureView(projection.Name, true)
+		if err != nil {
+			// skip on demand feature views
+			continue
+		}
+		for entityName, _ := range view.Entities {
+			entity := entitiesByName[entityName]
+			joinKeyTypes[entity.JoinKey] = int32(entity.ValueType.Number())
```

Comment:
`GetEntityTypesMapByFeatureService` is a function called from Python. Gopy doesn't understand that ValueType is an alias for int32, so packing and unpacking it will be additional headache, not sure that it will make it more readable.

Reply:
I'm just saying that using native types is easier in some cases.